### PR TITLE
Breadcrumbs

### LINKED ISSUE #157: Question about Breadcrumbs and Navigation component?
@illright I notice that you have the breadcrumbs component on the doc site but is not available from the library. Same to the navigation bar component; is there a particular reason that not supporting it?

Many thanks for your time.

### PR DESCRIPTION
**Pull request by aabounegm**
Implements the `Breadcrumbs` component, taken from the docs and generalized to allow any number of segments, as well as customizing the segments and separators.

Closes #157

### CODE REVIEW

**Comment by illright** [comment]
The docs don't seem to work when you click "docs" from the homepage

**Comment by aabounegm** [comment]
And that's why I hate `null`

File: docs/src/routes/docs/components/breadcrumbs.svx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+### `separator` slot
+
+If a string separator is not enough, use this slot to add a custom component separator (and ignore the prop).
+
+## SCSS Variables {#scss-variables}
+
+| Name | Description | Default |
+| ---- | ----------- | ------- |
+| **`$thin-font-weight`** | Description. | `300` |
```
Comment (illright) [inline docs/src/routes/docs/components/breadcrumbs.svx:101]:
```suggestion
| **`$thin-font-weight`** | The font size of the separator and the segments without `href`. | `300` |
```

File: attractions/breadcrumbs/breadcrumbs.scss [hunk 2]
```diff
@@ -0,0 +1,13 @@
+@use '../_variables' as vars;
+
+nav {
+  display: flex;
+  align-items: center;
+  flex: 1;
+  margin: 0 2em;
+  font-weight: vars.$thin-font-weight;
+
+  :global(.node) {
```
Comment (illright) [inline attractions/breadcrumbs/breadcrumbs.scss:10]: This might leak inside the custom segments
```suggestion
  & > :global .node {
```

**Review by illright — CHANGES_REQUESTED** [review]

File: docs/src/routes/docs/components/breadcrumbs.svx (hunk 1 above)
Comment (aabounegm) [reply docs/src/routes/docs/components/breadcrumbs.svx:101]: Whoops. That's what happens when you just copy the template :D

File: attractions/breadcrumbs/breadcrumbs.scss (hunk 2 above)
Comment (aabounegm) [reply attractions/breadcrumbs/breadcrumbs.scss:10]: That was kinda the point, to be able to give the `node` class to a slot

Comment (illright) [reply attractions/breadcrumbs/breadcrumbs.scss:10]: Yeah, this still provides this option, but only for direct children, which is where you'd want it

**Review by illright — APPROVED** [review]

**Pull request merged**